src/modules/design/diagram-detail/diagram-detail.ts

```diff
@@ -350,6 +374,28 @@ export class DiagramDetail {
     }
   }
 
+  private async _openDirectory(): Promise<string> {
+    const canNotReadFromFileSystem: boolean = !this._canReadFromFileSystem();
+    if (canNotReadFromFileSystem) {
+      return;
+    }
+
+    this._ipcRenderer.send('save_single_diagram');
```

Comment:
We use kebab case not snake case for ipcRenderer events

```suggestion
    this._ipcRenderer.send('save-single-diagram');
```